pkgs/development/tools/jdt-language-server/default.nix

```diff
@@ -0,0 +1,69 @@
+{ lib
+, stdenv
+, fetchurl
+, makeWrapper
+, jdk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "jdt-language-server";
+  version = "0.67.0";
+  timestamp = "202012170459";
+
+  src = fetchurl {
+    url = "https://download.eclipse.org/jdtls/milestones/${version}/jdt-language-server-${version}-${timestamp}.tar.gz";
+    sha256 = "11qzwf2l737hdv1g7bfcrmdxcrl7g9z2k9nhy9lknlabmhq4mcdl";
+  };
+
+  sourceRoot = ".";
+
+  buildInputs = [
+    jdk
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    # Copy jars
+    install -D -t $out/share/java/plugins/ plugins/*.jar
+
+    # Copy config directories for linux and mac
+    mkdir -p $out/share/config_{linux,mac}/
+    install -D config_linux/* $out/share/config_linux/
+    install -D config_mac/* $out/share/config_mac/
+
+    # Get latest version of launcher jar
+    # e.g. org.eclipse.equinox.launcher_1.5.800.v20200727-1323.jar
+    launcher="$(ls $out/share/java/plugins/org.eclipse.equinox.launcher_* | sort -V | tail -n1)"
+
+    # jdt-language-server needs to write to the config location, so we create a temp
+    # directory each time and copy the config_{linux,mac}/ folders there
+    runtime_dir=''${XDG_CACHE_HOME:-$HOME/.cache}/jdt-language-server
+
+    makeWrapper ${jdk}/bin/java $out/bin/jdt-language-server \
+      --run "mkdir -p $runtime_dir" \
+      --run "cp -r $out/share/config_* $runtime_dir" \
+      --run "chmod -R 1777 $runtime_dir" \
+      --add-flags "-Declipse.application=org.eclipse.jdt.ls.core.id1" \
+      --add-flags "-Dosgi.bundles.defaultStartLevel=4" \
+      --add-flags "-Declipse.product=org.eclipse.jdt.ls.core.product" \
+      --add-flags "-Dlog.level=ALL" \
+      --add-flags "-noverify" \
+      --add-flags "-jar $launcher" \
+      --add-flags "-configuration $runtime_dir/config_linux" \
+      --add-flags "-data $runtime_dir/data\$PWD" \
+      --add-flags "--add-modules=ALL-SYSTEM" \
+      --add-flags "--add-opens java.base/java.util=ALL-UNNAMED" \
+      --add-flags "--add-opens java.base/java.lang=ALL-UNNAMED"
```

Comment:
I would :+1: if you could comment each flag.

Reply:
I commented the flags that the user may want to change. The rest of them seem to be required and I don't see documentation in the project for what exactly they do (e.g. `-Declipse.application`, `-Dosgi.bundles.defaultStartLevel`). [`-noverify` is deprecated](https://bugs.openjdk.java.net/browse/JDK-8218003), but I don't see any documentation of why it should be set.